android/services/mongodb-remote-perftests/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/performance/SyncR2LOnlyPerformanceTestDefinitions.kt

```diff
@@ -0,0 +1,269 @@
+                doTestSyncPass(testHarness, runId, it)
+            }
+        }
+
+        private fun doTestSyncPass(
+            testHarness: SyncPerformanceIntTestsHarness,
+            runId: ObjectId,
+            pctOfDocsWithChangeEvents: Double
+        ) {
+            val testName = "testR2L_SyncPass_${pctOfDocsWithChangeEvents}_PctDocsChanged"
+
+            // Local variable for the number of docs updated in the test
+            // This should change for each iteration of the test.
+            var numberOfChangedDocs: Int? = null
+
+            testHarness.runPerformanceTestWithParams(
+                testName, runId,
+                beforeEach = { ctx, numDocs: Int, docSize: Int ->
+                    val sync = ctx.testColl.sync()
+
+                    // Generate the documents that are to be synced via R2L and insert remotely
+                    val ids = SyncPerformanceTestUtils.insertToRemote(
+                        ctx, numDocs, docSize
+                    )
+
+                    // If sync fails for any reason, halt the test
+                    Tasks.await(ctx.testColl.sync().configure(
+                        DefaultSyncConflictResolvers.remoteWins(),
+                        null,
+                        ExceptionListener { id, ex ->
+                            testHarness.logMessage(
+                                "unexpected sync error with id " +
+                                "$id: ${ex.localizedMessage}")
+                            error(ex)
+                        }
+                    ))
+
+                    // Sync on the ids inserted remotely
+                    Tasks.await(sync.syncMany(*(ids!!.toTypedArray())))
+
+                    // Halt the test if the sync pass failed
+                    if (!SyncPerformanceTestUtils.doSyncPass(ctx)) {
+                        error("sync pass failed")
+                    }
+
+                    // Verify that the test did indeed synchronize the provided documents locally
+                    val numSyncedIds = Tasks.await(sync.syncedIds).size
+                    val numLocalDocs = Tasks.await(sync.count())
+
+                    SyncPerformanceTestUtils.assertIntsAreEqualOrThrow(
+                        numSyncedIds, numDocs, "Number of Synced Ids")
+                    SyncPerformanceTestUtils.assertIntsAreEqualOrThrow(
+                        numLocalDocs.toInt(), numDocs, "Number of Local Documents")
+
+                    // Disconnect the DataSynchronizer and wait
+                    // for the underlying streams to close
+                    ctx.testNetworkMonitor.connectedState = false
+                    while (ctx.testDataSynchronizer.areAllStreamsOpen()) {
+                        testHarness.logMessage("waiting for streams to close")
+                        Thread.sleep(1000)
+                    }
+
+                    // Randomly sample a percentage of the documents
+                    // that will be locally updated
```

Comment:
[minor] update comment to clarify that this is randomly sampling a percentage of the documents to be **remotely** updated.